e2e/rbd.go

```diff
@@ -1023,17 +1025,114 @@ var _ = Describe("RBD", func() {
+				err = createRBDStorageClass(f.ClientSet, f, "", nil, nil, deletePolicy)
 				if err != nil {
 					e2elog.Failf("failed to create storageclass with error %v", err)
 				}
 			})
 
+			By("create rbd snapshots in different pool", func() {
+				// snapshot beta is only supported from v1.17+
+				if k8sVersionGreaterEquals(f.ClientSet, 1, 17) {
+					err := deleteResource(rbdExamplePath + "snapshotclass.yaml")
+					if err != nil {
+						e2elog.Failf("failed to delete snapshotclass with error %v", err)
+					}
+					snapshotPool := "snapshot-test"
+					// create pool for snapshots
+					err = createPool(f, snapshotPool)
+					if err != nil {
+						e2elog.Failf("failed to create pool %s with error %v", snapshotPool, err)
+					}
+					// create snapshotclass to with pool option
+					param := map[string]string{
+						"pool": snapshotPool,
+					}
+					err = createRBDSnapshotClass(f, param)
+					if err != nil {
+						e2elog.Failf("failed to create snapshotclass with error %v", err)
+					}
+					err = validateSnapshotInDifferentPool(f, snapshotPool, "", defaultRBDPool)
+					if err != nil {
+						e2elog.Failf("failed to validate snapshot in different pool with error %v", err)
+					}
+
+					err = deletePool(snapshotPool, false, f)
+					if err != nil {
+						e2elog.Failf("failed to delete pool %s with error %v", snapshotPool, err)
+					}
+					err = deleteResource(rbdExamplePath + "snapshotclass.yaml")
+					if err != nil {
+						e2elog.Failf("failed to delete snapshotclass with error %v", err)
+					}
+					err = createRBDSnapshotClass(f, nil)
+					if err != nil {
+						e2elog.Failf("failed to create snapshotclass with error %v", err)
+					}
+				}
+			})
+
+			By("create rbd clones in different pool", func() {
+				// snapshot beta is only supported from v1.17+
+				if k8sVersionGreaterEquals(f.ClientSet, 1, 17) {
+					err := deleteResource(rbdExamplePath + "snapshotclass.yaml")
+					if err != nil {
+						e2elog.Failf("failed to delete snapshotclass with error %v", err)
+					}
+					clonePool := "clone-test"
+					// create pool for clones
+					err = createPool(f, clonePool)
+					if err != nil {
+						e2elog.Failf("failed to create pool %s with error %v", clonePool, err)
+					}
+					// create snapshotclass to with pool option
+					param := map[string]string{
+						"pool": defaultRBDPool,
+					}
+					err = createRBDSnapshotClass(f, param)
+					if err != nil {
+						e2elog.Failf("failed to create snapshotclass with error %v", err)
+					}
+					cloneSC := "clone-storageclass"
+					param = map[string]string{
+						"pool": clonePool,
+					}
+					// create new storageclass to with new pool
+					err = createRBDStorageClass(f.ClientSet, f, cloneSC, nil, param, deletePolicy)
+					if err != nil {
+						e2elog.Failf("failed to create storageclass with error %v", err)
+					}
+					err = validateSnapshotInDifferentPool(f, defaultRBDPool, cloneSC, clonePool)
```

Comment:
The function here is validateSnapshotInDifferentPool and err says `failed to validate clone in different pool`

Reply:
The function `validateSnapshotInDifferentPool` refers to creating snapshots in a different pool. It contains parameters (cloneSC,clonePool in this case) that allow the restore operation of the created snapshots in a different pool to verify working across multiple pools.
The error message `failed to validate clone in different pool` refers to the failure of the complete test case which is `create rbd clones in different pool`. Let me know if err message is not apt :)

Reply:
yeah I get that, if we are looking for highlevel test function reference, we could log it as `failed to create rbd clones in different pool` too.  I am fine with current form though.